pkg/operator/csidriveroperator/csioperatorclient/azure-file.go

```diff
@@ -13,6 +13,19 @@ const (
 	envAzureFileDriverImage         = "AZURE_FILE_DRIVER_IMAGE"
 )
 
+func IsNotAzueStackCloud(status *configv1.InfrastructureStatus) bool {
+	if status == nil {
+		return false
+	}
+	// Azure File is not supported on Azure StackHub
+	// https://learn.microsoft.com/en-us/azure-stack/user/azure-stack-acs-differences?view=azs-2206#cheat-sheet-storage-differences
+	// TODO: remove this StatusFilter if Azure File gets support in the future
+	if status.PlatformStatus.Azure.CloudName == configv1.AzureStackCloud {
+		return false
```

Comment:
This looks OK for new installations, but what about the existing ones? Suddenly CSO will stop managing the operator. After upgrade, no images will be updated in the operator, running both the operator and CSI driver with potentially insecure binaries.

It should return `false` only when Azure File operator is not running yet, e.g. when its `clustercsidriver` does not exist. It's not a great detection mechanism, but it's better than nothing. Unfortunately, it won't be as simple as filing `StatusFilter`.

Reply:
Good point, but isn't the same issue in `StatusFilter` for IBM VPC? Anyway, I posted a new version that checks the `clustercsidriver` and stops managing Azure File only if it's not present. I had more ideas about the implementation but this seemed ok to me. 

What I'm not sure about is the removal of Azure File in existing StackHub clusters. I just turned off CSO before removing related Azure File objects for testing (see verification below). Do we have some better recommended and well known way for the removal? If not I suppose a release note might be useful. Looking at the bug report it seems pretty much broken so there should not be anything provisioned in existing clusters by Azure File.

Reply:
> but isn't the same issue in StatusFilter for IBM VPC? 

On IBM VPC we check for IBM management cluster and that does not change during upgrade. It should not happen that `isInstalled` gets `true` there.

Reply:
Ack, warning added.